druid-shell/src/platform/mac/application.rs

```diff
@@ -85,4 +87,17 @@ impl Application {
             }
         }
     }
+
+    pub fn get_locale() -> String {
+        unsafe {
+            let nslocale_class = class!(NSLocale);
+            let locale: id = msg_send![nslocale_class, currentLocale];
+            let ident: id = msg_send![locale, localeIdentifier];
+            let mut locale = util::from_nsstring(ident);
+            if let Some(idx) = locale.chars().position(|c| c == '@') {
```

Comment:
This is not Unicode-correct, the result of position is a code point count, the truncation expects utf-8 code units. But the locale string itself will always be ascii, so this is a nit.

Reply:
yes this was a total hack when it was written, just trying to get something working.